src/Xrefcheck/Config/Default.hs

```diff
@@ -56,6 +56,35 @@ networking:
   # a "429 Too Many Requests" response.
   maxRetries: 3
 
+  # Maximum number of links that can be followed in a single redirect
+  # chain.
+  #
+  # The link is considered as invalid if the limit is exceeded.
+  maxRedirectFollows: 10
+
+  # Rules to override the redirect behavior for external references that
+  # match, where
+  #   - 'from' is a regular expression for the source link in the redirection.
+  #     Its absence means that every link matches.
+  #   - 'to' is a regular expression for the target link in the redirection.
+  #     Its absence also means that every link matches.
+  #   - 'on' accepts 'temporary', 'permanent' or a specific redirect HTTP code.
+  #     Its absence also means that every response code matches.
+  #   - 'outcome' accepts 'valid', 'invalid' or 'follow'. The last one follows
+  #      the redirect by applying the same configuration rules.
```

Comment:
Hm, what is meant by "applying the same configuration rules"? That, when a rule with `follow` matches, we process the next link with same configured list of rules? But we apply them always, why should we mention it separately?

Reply:
Previously, redirects were automatically followed by the HTTP client, so none of our behaviours were applied to each step, but to the entire chain. For example:
- Timeout was applied to the entire chain.
- Retry logic was applied to the entire chain.
- Ignore check was only applied to the initial link.

This new feature allows to explicitly follow redirects, and we have decided to apply the same configuration rules at each step of the chain. But we are still applying everything else to the entire chain, and we plan to review it in other issues like #260.

Reply:
Yeah. It may sound a bit confusing (due to how obvious it seems), but the behaviour should be clarified.

Probably if we add something like:

> so for instance, exclusion rules would apply to the following links

it would sound more benign, more like a hint about a possible trick rather than an obvious statement, and would still make the exact behaviour clear.